FRAuthenticator/FRAuthenticatorTests/E2ETests/FRAClient/FRAClientTests.swift

```diff
@@ -800,7 +798,160 @@ class FRAClientTests: FRABaseTests {
+
+        guard let account2 = authenticatorManager.getAccount(identifier: "ForgeRock-pushtestuser2"),
+           let mechanism2 = account2.mechanisms.first as? PushMechanism else {
+           XCTFail("Failed to retrieve PushMechanism")
+           return
+        }
+        //  Update Mechanism with UUID
+        mechanism2.mechanismUUID = "862BCE9A-A74A-33D2-181D-77F6B54C3A10"
+        XCTAssertTrue(FRAClient.storage.setMechanism(mechanism: mechanism2))
+        
+        //  Fake first remote-message receive action
+        var payload: [String: Any] = [:]
+        var aps: [String: Any] = [:]
+        aps["messageId"] = "AUTHENTICATE:64e909a2-84db-4ee8-b244-f0dbbeb8b0ff1589151035455"
+        aps["content-available"] = true
+        aps["alert"] = "Login attempt from user at ForgeRockSandbox"
+        aps["data"] = "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJjIjoicGtJZm1rZDFSbDNJWnFhSmNZSUhLUzJic2wvZnhiWGNPYTEvNE83Z2pMTT0iLCJ0IjoiMTIwIiwidSI6Ijc1OUFDRTlELUM2NEItNDNFNi05ODFELTk3RjdCNTRDM0IwMSIsImwiOiJZVzFzWW1OdmIydHBaVDB3TVE9PSJ9.o_F0jvSiQlcpInyexi5ED4pjKNHxoHknPIsowYbpyYQ"
+        aps["sound"] = "default"
+
+        payload["aps"] = aps
+
+        guard let _ = FRAPushHandler.shared.application(UIApplication.shared, didReceiveRemoteNotification: payload) else {
+          XCTFail("Failed to parse notification payload and construct PushNotification object")
+          return
+        }
+
+        var pushAccount = authenticatorManager.getAccount(identifier: "ForgeRockSandbox-pushtestuser")
+        var pushMechanism = pushAccount?.mechanisms.first as? PushMechanism
+
+        XCTAssertEqual(pushMechanism?.notifications.count, 1)
+        XCTAssertEqual(storageClient.defaultStorageClient.accountStorage.allItems()?.count, 2)
+        XCTAssertEqual(storageClient.defaultStorageClient.mechanismStorage.allItems()?.count, 2)
+        XCTAssertEqual(storageClient.defaultStorageClient.notificationStorage.allItems()?.count, 1)
+
+        //  Fake second remote-message receive action
+        payload = [:]
+        aps = [:]
+        aps["messageId"] = "AUTHENTICATE:0666696b-859d-4565-b069-f13c800c5e3c1589151071515"
+        aps["content-available"] = true
+        aps["alert"] = "Login attempt from user at ForgeRockSandbox"
+        aps["data"] = "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJjIjoiK2ZreklzbG1PN01GcHYzZEppQVY4Sm5yRzFra0hUdDA2NkRaaEcxamJwbz0iLCJ0IjoiMTIwIiwidSI6Ijc1OUFDRTlELUM2NEItNDNFNi05ODFELTk3RjdCNTRDM0IwMSIsImwiOiJZVzFzWW1OdmIydHBaVDB3TVE9PSJ9.1zuEgWy1gN7QAaJscGKgAqGxt_58Ad5HAZjOe8PLMBo"
+        aps["sound"] = "default"
+
+        payload["aps"] = aps
+
+        guard let _ = FRAPushHandler.shared.application(UIApplication.shared, didReceiveRemoteNotification: payload) else {
+          XCTFail("Failed to parse notification payload and construct PushNotification object")
+          return
+        }
+
+
+        pushAccount = authenticatorManager.getAccount(identifier: "ForgeRockSandbox-pushtestuser")
```

Comment:
Not quite sure on this, perhaps I might forget on something, is there any reason that we keep reading the object from the storage?

Reply:
Same as above.